Libraries/LibJS/Runtime/GlobalObject.cpp

```diff
@@ -247,8 +247,13 @@ JS_DEFINE_NATIVE_FUNCTION(GlobalObject::parse_float)
     auto string = vm.argument(0);
 
     // OPTIMIZATION: We can skip the number-to-string-to-number round trip when the value is already a number.
-    if (string.is_number())
+    if (string.is_number()) {
+        // Special case for negative zero - it should become positive zero
+        if (string.as_double() == 0 && std::signbit(string.as_double()))
```

Comment:
We try to avoid using `std` functions where possible. Could you use `string.is_negative_zero()` here instead?